### DISCUSSION TITLE
[✨] nonce on styles

### DISCUSSION THREAD

**Discussion by tzdesign** [QwikDev/qwik-evolution#264] [Proposals For Qwik]
### Is your feature request related to a problem?

No

### Describe the solution you'd like

I know that unsafe-inline in in CSP for style tags is totally ok, but I wish to have the possibility to have the nonce there anyway.

The style-tags should render with nonce if the shared map has a value for @nonce set like:

```ts
ev.sharedMap.set('@nonce', nonce);
```

I would do this myself, but ```_appendHeadStyle``` is so basic without additional data than ID and style, that I have no idea how to get the global context here.

### Describe alternatives you've considered

unsafe-inline is ok, also suggested by google.

If you do online-banking software or similar high security apps, It would be good to have all tags secured by nonce.

**Comment by manucorporat** [comment]
is the nonce also needed for CSR rendering? or only for SSR rendering?

**Comment by manucorporat** [comment]
This should make it easier to implement:
https://github.com/BuilderIO/qwik/pull/4478

**Comment by tzdesign** [comment]
Thanks @manucorporat 
This looks awesome.

Let me have a look at it after my vacation next week.

**Comment by gioboa** [comment]
is it still valid?

**Comment by tzdesign** [comment]
Still valid. Nobody is using prober CSP I guess.

**Comment by gioboa** [comment]
We moved this issue to `qwik-evolution` repo to create a RFC discussion for this.
Here is our [Qwik RFC process](https://github.com/QwikDev/qwik-evolution?tab=readme-ov-file#qwik-rfc-process) thanks.

**Comment by wmertens** [comment]
@tzdesign how do you use prober csp? :) and also, is this fixed now?
